test/native/symbolsLinuxTest.cpp

```diff
@@ -33,4 +33,19 @@ TEST_CASE(ResolveFromRela_dyn_R_ABS64) {
     ASSERT_RESOLVE(im_pthread_exit);
 }
 
+TEST_CASE(VirtAddrDifferentLoadAddr) {
+    void* result = dlopen("libvaddrdif.so", RTLD_NOW);
+    if (!result) {
+        printf("%s\n", dlerror());
+    }
+    ASSERT(result);
+    Profiler::instance()->updateSymbols(false);
+
+    PerfEventType* event_type = PerfEventType::forName("vaddrdif_square");
```

Comment:
`Profiler::resolveSymbol()` will be more reliable.
First, because `PerfEventType::forName` attempts to use `dlsym`, which will return a correct address even when profiler's parser breaks. Second, because `PerfEventType::forName` returns an address of a breakpoint instead of an entry address of a function. They [differ](https://github.com/async-profiler/async-profiler/blob/49cca40e472b5156de4b2ab112429991d42088ac/src/arch.h#L118-L120) on PPC.